core/src/main/java/io/github/kaktushose/jdac/JDACBuilder.java

```
@@ -238,73 +302,78 @@ public JDACBuilder shutdownJDA(boolean shutdown) {
                     new InteractionRegistry(
-                            new Validators(validators()),
-                            i18n,
-                            localizeCommands() ? i18n.localizationFunction() : (_) -> Map.of(),
-                            descriptor()
+                            new Validators(loader.get(VALIDATOR)),
+                            loader.get(I18N),
+                            loader.get(LOCALIZE_COMMANDS) ? loader.get(I18N).localizationFunction() : (_) -> Map.of(),
+                            loader.get(DESCRIPTOR)
                     ),
-                    new TypeAdapters(typeAdapters(), i18n),
-                    expirationStrategy(),
-                    controllerInstantiator(),
-                    embeds(messageResolver),
-                    i18n,
-                    messageResolver,
-                    globalReplyConfig(),
-                    globalCommandConfig()
+                    new TypeAdapters(loader.get(TYPE_ADAPTER), loader.get(I18N)),
+                    loader.get(EXPIRATION_STRATEGY),
+                    loader.get(INTERACTION_CONTROLLER_INSTANTIATOR),
+                    loader.get(EMBEDS),
+                    loader.get(I18N),
+                    loader.get(MESSAGE_RESOLVER),
+                    loader.get(GLOBAL_REPLY_CONFIG),
+                    loader.get(GLOBAL_COMMAND_CONFIG)
             );
 
             JDACommands jdaCommands = new JDACommands(
                     frameworkContext,
-                    context(),
-                    guildScopeProvider(),
-                    shutdownJDA()
+                    loader.get(JDA_CONTEXT),
+                    loader.get(GUILD_SCOPE_PROVIDER),
+                    loader.get(SHUTDOWN_JDA)
             );
 
-            jdaCommands.start(mergedClassFinder());
+            jdaCommands.start(loader.get(MERGED_CLASS_FINDER));
             return jdaCommands;
         } catch (JDACException e) {
```

shouldn't we catch all exceptions here?

---

Well, I didn't change the logic here. Do we want to shut down jda in the case of any exception not just "controlled" JDAExceptions?

---

I could swear that I've once catched all exceptions anywhere. Also I believe that JDA might throw IllegalArgumentExceptions during startup. I'll investigate this, if you want to merge you can make this an issue so I dont forget about it

---

ah okay. We wrap JDAs IllegalArgumentExceptions and then rethrow a JDA-Commands exception